integration/tests/access/unstaked_node_test.go

```diff
@@ -40,23 +40,18 @@ func TestUnstakedAccessSuite(t *testing.T) {
 }
 
 func (suite *UnstakedAccessSuite) TearDownTest() {
-	// avoid nil pointer errors for skipped tests
-	if suite.cancel != nil {
-		defer suite.cancel()
-	}
-	if suite.net != nil {
-		suite.net.Remove()
-	}
+	defer suite.cancel()
+	suite.net.Remove()
 }
 
 func (suite *UnstakedAccessSuite) SetupTest() {
+	suite.ctx, suite.cancel = context.WithCancel(context.Background())
```

Comment:
creating the context here makes sure that `suite.ctx` and `suite.cancel` are never nil when `TearDownTest` occurs.

Reply:
tbh it wouldn't matter - since there is no return before the `suite.buildNetworkConfig()` but I still feel that setting the context upfront makes it more readable.